source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -545,16 +716,22 @@ void InjaTransformer::transform(Http::RequestOrResponseHeaderMap &header_map,
     }
   }
   // get the extractions
-  std::unordered_map<std::string, absl::string_view> extractions;
+  std::unordered_map<std::string, std::string> extractions;
   if (advanced_templates_) {
     extractions.reserve(extractors_.size());
   }
 
   for (const auto &named_extractor : extractors_) {
     const std::string &name = named_extractor.first;
+    const ExtractionApi::Mode mode = named_extractor.second.mode();
     if (advanced_templates_) {
-      extractions[name] =
+      if (mode == ExtractionApi::REPLACE_ALL || mode == ExtractionApi::SINGLE_REPLACE) {
```

Comment:
maybe just use a switch/case with all modes here?

Reply:
If we move to switch we can also use the CORRUPT_ENUM from upstream if need be